src/Microsoft.Health.Dicom.Core/Features/Store/StoreDatasetValidator.cs

```diff
@@ -365,14 +365,14 @@ private void ValidateWithoutNullPadding(string value, DicomElement de, IReadOnly
     }
 
     private static KeyValuePair<string, object>[] TelemetryDimension(DicomItem item, bool isIndexableTag) =>
-        TelemetryDimension(item.Tag, isIndexableTag);
+        TelemetryDimension(item.Tag, item.ValueRepresentation, isIndexableTag);
 
-    private static KeyValuePair<string, object>[] TelemetryDimension(DicomTag tag, bool isIndexableTag) =>
+    private static KeyValuePair<string, object>[] TelemetryDimension(DicomTag tag, DicomVR vr, bool isIndexableTag) =>
         new[]
         {
-            new KeyValuePair<string, object>("TagKeyword", tag.DictionaryEntry.Keyword),
-            new KeyValuePair<string, object>("VR", tag.GetDefaultVR().ToString()),
-            new KeyValuePair<string, object>("Tag", tag.ToString()),
+            new KeyValuePair<string, object>("TagKeyword", tag?.DictionaryEntry?.Keyword),
+            new KeyValuePair<string, object>("VR", vr?.ToString()),
```

Comment:
I think if you just do tag.GetDefaultVR() instead of ToString() that is enough right. We dont want all this change?